pandas/core/frame.py

```diff
@@ -6891,15 +6913,39 @@ def sort_values(
         3   48hr     40
         4   96hr     50
         >>> from natsort import index_natsorted
+        >>> index_natsorted(df["time"])
+        [0, 3, 2, 4, 1]
         >>> df.sort_values(
-        ...     by="time", key=lambda x: np.argsort(index_natsorted(df["time"]))
+        ...     by="time",
+        ...     key=lambda x: np.argsort(index_natsorted(x)),
         ... )
             time  value
         0    0hr     10
         3   48hr     40
         2   72hr     30
         4   96hr     50
         1  128hr     20
+
+        If sorted indices are not needed, you may simplify the procedure as follows:
+
+        >>> from natsort import natsort_keygen
+        >>> df.sort_values(
+        ...     by="time",
+        ...     key=natsort_keygen(),
+        ... )
+            time  value
+        0    0hr     10
+        3   48hr     40
+        2   72hr     30
+        4   96hr     50
+        1  128hr     20
+
+        .. warning::
```

Comment:
I understand why you're writing this, but feels like a bit out of topic in the docstring for pandas' `sort_values`. Personally, I found that the last example may be useful for natsort users, but it doesn't add much to explain the `sort_values` method, which is the point of this PR.

If you don't think natsort is reliable, maybe better to just keep the example with str.lower.

Reply:
Thanks @datapythonista, I see your point.
Meanwhile, I do see the value of showing how `natural sorting` can be added to `sort_values`. This example was how I learned about `natsort` in the first place.

By the way, the issue that I am currently warning about is actually originating from `pd.MultiIndex` (and not `natsort` per se) where `pd.MultiIndex` silently (i.e., no warning, no errors) creates a single-level index if the provided tuples do not have the same dimensions (see the linked issue above).

In short, would be happy to follow your suggestion. I will reduce the examples, and provide a small reference to `natsort` for anyone who might need it. Would you agree with this?

Reply:
I think showing an example with `key` is good, and if natsort has a function we can already use and you think it's a relevant library for users I'm surely happy to have it. My point was that we had two examples showing pretty much the same pandas functionality, and showing different natsort functionalities, the second requiring a warning because it was buggy or misleading. So, probably better to remove the second example. Users reading our page are likely to be interested in pandas functionality  if they discover natsort that's great, but for more natsort examples they can go to natsort documentation.

In summary, happy to have natsort used, but IMHO better not to show advanced natsort things that aren't relevant to pandas users not interested in natsort, or that require a warning. Does it make sense?

Reply:
Of course. Sure, lets go with a single example then.
Please see if you agree with the latest update.

Thanks :)